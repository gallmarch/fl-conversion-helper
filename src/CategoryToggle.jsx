import React from 'react';

export default function CategoryToggle(props) {
  if (props.state) {
    return <span className="contract">-</span>;
  }
  return <span className="expand">+</span>;
}
