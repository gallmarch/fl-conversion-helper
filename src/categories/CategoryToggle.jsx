import React from 'react';
import PropTypes from 'prop-types';

export default function CategoryToggle(props) {
  if (props.state) {
    return <span className="contract">-</span>;
  }
  return <span className="expand">+</span>;
}

CategoryToggle.propTypes = {
  state: PropTypes.bool.isRequired,
};
