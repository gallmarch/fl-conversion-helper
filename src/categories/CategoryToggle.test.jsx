import React from 'react';
import renderer from 'react-test-renderer';

import CategoryToggle from './CategoryToggle';

describe('CategoryToggle', () => {
  it('matches its snapshot', () => {
    const componentTrue = renderer.create(<CategoryToggle state={true} />);
    expect(componentTrue).toMatchSnapshot();

    const componentFalse = renderer.create(<CategoryToggle state={false} />);
    expect(componentFalse).toMatchSnapshot();

    const componentUndefined = renderer.create(<CategoryToggle />);
    expect(componentUndefined).toMatchSnapshot();
  });
});
