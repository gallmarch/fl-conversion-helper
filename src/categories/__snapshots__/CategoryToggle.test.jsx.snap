// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryToggle matches its snapshot 1`] = `
<span
  className="contract"
>
  -
</span>
`;

exports[`CategoryToggle matches its snapshot 2`] = `
<span
  className="expand"
>
  +
</span>
`;

exports[`CategoryToggle matches its snapshot 3`] = `
<span
  className="expand"
>
  +
</span>
`;
